File: 1743/1743.go
func restoreTheArrayFromAdjacentPai(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := restoreTheArrayFromAdjacentPai(root.Left)
	r := restoreTheArrayFromAdjacentPai(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1743/1743.cs
public class Solution {
    public int RestoreTheArrayFromAdjacentPai(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(RestoreTheArrayFromAdjacentPai(root.left), RestoreTheArrayFromAdjacentPai(root.right));
    }
}